src/arch/x86/lib/cpu_performance/cpu_info.c

```diff
@@ -21,10 +21,12 @@ static void set_cpu_features(struct cpu_info *info) {
 					: "=g"(reg)
 					:
 					: "%eax", "%ebx", "%ecx", "%edx");
-
-	info->FPU = reg & 0x1;
-	info->MMX = (reg >> 23) & 0x1; 
-	info->SSE = (reg >> 25) & 0x1;
+	strcpy(info->feature[2].name, "FPU");
```

Comment:
Bad thing here is that programmer should track feature numbers manually.
What if we move some piece of code to another function or another file?
Some indices may end up being unused, other may be less than `current_cpu.num_features` if programmer forgets to increment it.
It's easy to track if you have 5 features, but, for example, `lscpu` prints 112 features for my CPU on my host machine, tracking this stuff may become painful :)

So a better solution would be tracking it within some simple handle, like
```
int set_cpu_feature(struct cpu_info *info, const char *name, unsigned val) {
   /* Set info->feature[info->num_features],
    * then increment info->num_features;
    * check for overflow etc */
}
```

Reply:
Done in the latest fixup commit!